flash/vision/classification/data.py

```diff
@@ -13,152 +13,33 @@
-
-        root/class_x/xxx.ext
-        root/class_x/xxy.ext
-        root/class_x/xxz.ext
-
-        root/class_y/123.ext
-        root/class_y/nsdf3.ext
-        root/class_y/asd932_.ext
-
-    Args:
-        root: Root directory path.
-        loader: A function to load a sample given its path.
-        extensions: A list of allowed extensions. both extensions
-            and is_valid_file should not be passed.
-        transform: A function/transform that takes in
-            a sample and returns a transformed version.
-            E.g, ``transforms.RandomCrop`` for images.
-        target_transform: A function/transform that takes
-            in the target and transforms it.
-        is_valid_file: A function that takes path of a file
-            and check if the file is a valid file (used to check of corrupt files)
-            both extensions and is_valid_file should not be passed.
-        with_targets: Whether to include targets
-        img_paths: List of image paths to load. Only used when ``with_targets=False``
-
-     Attributes:
-        classes (list): List of the class names sorted alphabetically.
-        class_to_idx (dict): Dict with items (class_name, class_index).
-        samples (list): List of (sample path, class_index) tuples
-        targets (list): The class_index value for each image in the dataset
-    """
-
-    def __init__(
-        self,
-        root: str,
-        loader: Callable,
-        extensions: Tuple[str] = IMG_EXTENSIONS,
-        transform: Optional[Callable] = None,
-        target_transform: Optional[Callable] = None,
-        is_valid_file: Optional[Callable] = None,
-        with_targets: bool = True,
-        img_paths: Optional[List[str]] = None,
-    ):
-        super(FlashDatasetFolder, self).__init__(root, transform=transform, target_transform=target_transform)
-        self.loader = loader
-        self.extensions = extensions
-        self.with_targets = with_targets
-
-        if with_targets:
-            classes, class_to_idx = self._find_classes(self.root)
-            samples = make_dataset(self.root, class_to_idx, extensions, is_valid_file)
-
-            if len(samples) == 0:
-                msg = "Found 0 files in subfolders of: {}\n".format(self.root)
-                if extensions is not None:
-                    msg += "Supported extensions are: {}".format(",".join(extensions))
-                raise RuntimeError(msg)
-
-            self.classes = classes
-            self.class_to_idx = class_to_idx
-            self.samples = samples
-            self.targets = [s[1] for s in samples]
-        else:
-            if not img_paths:
-                raise MisconfigurationException(
-                    "`FlashDatasetFolder(with_target=False)` but no `img_paths` were provided"
-                )
-            self.samples = img_paths
-
-    def _find_classes(self, dir):
+    @staticmethod
+    def _find_classes(dir):
```

Comment:
```suggestion
    def _get_classes(dir):
```
Prefix consistent with `_get_predicting_files`